### PR TITLE
fix #3051: fullscreen switching with CTRL+ALT+ENTER doesn't work any …

### LINKED ISSUE #3051: CTRL+ALT+ENTER doesn't work any more in xfreerdp
this commit broke full screen switching CTRL+ALT+ENTER for me:

https://github.com/FreeRDP/FreeRDP/commit/a9e3368d4473e146c54ede37e3edd331e6e68d59

### PR DESCRIPTION
**Pull request by abma**
KeyboardState is defined as:

BOOL KeyboardState[256];

we can't store keysym there, we must use bool

not sure if the fix is correct, the alternative would be to define KeyboardState as

int KeyboardState[256];

### CODE REVIEW

File: client/X11/xf_keyboard.c [hunk 1]
```diff
@@ -179,7 +179,7 @@ void xf_keyboard_release_all_keypress(xfContext* xfc)
 
 	for (keycode = 0; keycode < ARRAYSIZE(xfc->KeyboardState); keycode++)
 	{
-		if (xfc->KeyboardState[keycode] != NoSymbol)
+		if (!xfc->KeyboardState[keycode])
```
Comment (hardening) [inline client/X11/xf_keyboard.c]: I think it's the opposite here as NoSymbol in the old code means FALSE now

**Comment by hardening** [comment]
@freerdp-bot can you test this, please ?

File: client/X11/xf_keyboard.c (hunk 1 above)
Comment (abma) [inline client/X11/xf_keyboard.c]: true, will change / test

**Comment by akallabeth** [comment]
@abma Nice catch, if you can fix the issue @hardening pointed out. Except this (blocking) remark +1

**Comment by abma** [comment]
as far as i can tell, it works for me...

**Pull request merged**